lib/HLSL/HLOperationLower.cpp

```diff
@@ -600,18 +600,30 @@ Value *TrivialBarrier(CallInst *CI, IntrinsicOp IOP, OP::OpCode opcode,
 Value *TranslateD3DColorToUByte4(CallInst *CI, IntrinsicOp IOP,
                                  OP::OpCode opcode,
                                  HLOperationLowerHelper &helper,  HLObjectOperationLowerHelper *pObjHelper, bool &Translated) {
-  hlsl::OP *hlslOP = &helper.hlslOP;
   IRBuilder<> Builder(CI);
   Value *val = CI->getArgOperand(HLOperandIndex::kUnaryOpSrc0Idx);
   Type *Ty = val->getType();
 
-  Constant *toByteConst = ConstantFP::get(Ty->getScalarType(), 255);
-  if (Ty != Ty->getScalarType()) {
-    toByteConst =
-        ConstantVector::getSplat(Ty->getVectorNumElements(), toByteConst);
+  bool isFXCCompatMode = CI->getModule()->GetHLModule().GetHLOptions().bFXCCompatMode;
+  // Use literal 255.001953 for scaling when FXC compat mode is enabled as
+  // FXC uses this value instead of 255 for some uknown reason.
```

Comment:
I'm not sure I agree with the rationale but here it is: https://stackoverflow.com/questions/52103720/why-does-d3dcolortoubyte4-multiplies-components-by-255-001953f

Reply:
Cool! didn't expect stackoverflow thread on this. :)

Reply:
The answer is by one of our ATG people